core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -254,6 +259,29 @@ private[spark] class DAGScheduler(
   private val blockManagerMasterDriverHeartbeatTimeout =
     sc.getConf.get(config.STORAGE_BLOCKMANAGER_MASTER_DRIVER_HEARTBEAT_TIMEOUT).millis
 
+  private val shuffleMergeResultsTimeoutSec =
+    sc.getConf.get(config.PUSH_BASED_SHUFFLE_MERGE_RESULTS_TIMEOUT)
+
+  private val shuffleMergeFinalizeWaitSec =
+    sc.getConf.get(config.PUSH_BASED_SHUFFLE_MERGE_FINALIZE_TIMEOUT)
+
+  // Since SparkEnv gets initialized after DAGScheduler, externalShuffleClient needs to be
+  // initialized lazily
+  private lazy val externalShuffleClient: Option[ExternalBlockStoreClient] =
+    if (pushBasedShuffleEnabled) {
+      val transConf = SparkTransportConf.fromSparkConf(sc.conf, "shuffle", 1)
+      val shuffleClient = new ExternalBlockStoreClient(transConf, env.securityManager,
+        env.securityManager.isAuthenticationEnabled(),
+        sc.conf.get(config.SHUFFLE_REGISTRATION_TIMEOUT))
+      shuffleClient.init(sc.conf.getAppId)
+      Some(shuffleClient)
+    } else {
+      None
+    }
+
+  private val shuffleMergeFinalizeScheduler =
+    ThreadUtils.newDaemonThreadPoolScheduledExecutor("shuffle-merge-finalizer", 8)
```

Comment:
So we decide to use a fixed number of threads instead of exposing a config?

Reply:
Yes, I don't think we need a config as of now. In future, if needed we can always add a config.